File: Backend/Online_Survey/DTOs/Respondent/ResponseViaSurveyId.cs
using System.Collections.Generic;

namespace Online_Survey.DTOs.Respondent
{
    public class OptionDTO
    {
        public string OptionText { get; set; }
    }

    public class QuestionDTO
    {
        public string QuestionText { get; set; }
        public string QuestionType { get; set; }

        public List<string> AnswerTexts { get; set; } // List of answer texts

        public List<OptionDTO> Options { get; set; }
    }

    public class ResponseViaSurveyId
    {
        public int ResponseId { get; set; }
        public int RespondentId { get; set; }
        public int SurveyId { get; set; }

        public List<QuestionDTO> QuestionList { get; set; }
    }
}

File: ClientSide/Online_Survey/src/app/shared/Models/ResponseViaSurveyId.ts
export interface OptionDTO {
    optionText: string;
  }
  
  export interface QuestionDTO {
    questionText: string;
    questionType: string;
    answerTexts: string[]; // List of answer texts
    options: OptionDTO[];
  }
  
  export interface ResponseViaSurveyId {
    responseId: number;
    respondentId: number;
    surveyId: number;
    questionList: QuestionDTO[];
  }
  